src/main/java/hudson/remoting/ExportTable.java

```diff
@@ -43,8 +40,11 @@
  * @author Kohsuke Kawaguchi
  */
 final class ExportTable {
-    private final Map<Integer,Entry<?>> table = new HashMap<Integer,Entry<?>>();
-    private final Map<Object,Entry<?>> reverse = new HashMap<Object,Entry<?>>();
+    /**
+     * Entry table map based on unique id.
+     */
+    private final Map<Integer, Entry<?>> etable = new IdentityHashMap<Integer, Entry<?>>();
```

Comment:
Huh?? `IdentityHashMap<Integer,T>` makes little sense. The keys are autoboxed `int`s; the `System.identityHashCode` is meaningless for these.

Did you perhaps mean to leave `table` along and just make `reserve` an `IdentityHashMap` with the same signature?